python/lib/modeldata/dmod/modeldata/data/object_store_dataset.py

```diff
@@ -314,24 +381,32 @@ def add_data(self, dataset_name: str, **kwargs) -> bool:
         """
         if dataset_name not in self.datasets:
             return False
-        elif 'file' in kwargs and 'directory' in kwargs:
+        elif data is not None:
+            result = self._client.put_object(bucket_name=dataset_name, data=data, length=len(data),
+                                             object_name=self._gen_dataset_serial_obj_name(dest))
+            return result.bucket_name == dataset_name
```

Comment:
what is `object_name` used for here?  is `result.bucket_name == dataset_name` supposed to be `result.bucket_name == object_name`?  If not, then why generate the name just to return?

Reply:
Well, ``object_name`` is the MinIO API param for the name of the object, within the bucket, that has/is the data being added.  This wasn't passing it the right value; it should have just been `object_name=dest`.  I'll fix that.  

What comes back from `put_object()` is an `ObjectWriteResult`.  I don't exactly remember why I picked that expression to return, but I am sure that the bucket corresponds to the dataset, not the object.

I may not be following completely with what you are asking, so this might be good to bring up in our next touch point.

Reply:
I think I miss-read the code the first time here and was seeing `object_name=self._gen_dataset_serial_obj_name(dest)` not a a function arg but as its own line of code, and wasn't sure why it was needed haha

Reply:
Once again, I wasn't reading indentation/line wrapping correctly...though it sounds like that may have helped uncover a different potential bug? Who knew???